iceoryx_posh/source/popo/wait_set.cpp

```diff
@@ -33,91 +32,68 @@ WaitSet::WaitSet(cxx::not_null<ConditionVariableData* const> condVarDataPtr) noe
 
 WaitSet::~WaitSet() noexcept
 {
-    detachAllConditions();
+    removeAllTrigger();
     /// @todo Notify RouDi that the condition variable data shall be destroyed
 }
 
-cxx::expected<WaitSetError> WaitSet::attachCondition(Condition& condition) noexcept
+void WaitSet::removeTrigger(const Trigger& trigger) noexcept
 {
-    if (!isConditionAttached(condition))
+    for (auto& currentTrigger : m_triggerVector)
     {
-        if (!m_conditionVector.push_back(&condition))
+        if (currentTrigger.isLogicalEqualTo(trigger))
         {
-            return cxx::error<WaitSetError>(WaitSetError::CONDITION_VECTOR_OVERFLOW);
-        }
-
-        condition.attachConditionVariable(this, m_conditionVariableDataPtr);
-    }
-
-    return iox::cxx::success<>();
-}
-
-void WaitSet::detachCondition(Condition& condition) noexcept
-{
-    if (!condition.isConditionVariableAttached())
-    {
-        return;
-    }
-
-    condition.detachConditionVariable();
-}
-
-void WaitSet::removeCondition(const Condition& condition) noexcept
-{
-    for (auto& currentCondition : m_conditionVector)
-    {
-        if (currentCondition == &condition)
-        {
-            m_conditionVector.erase(&currentCondition);
+            m_triggerVector.erase(&currentTrigger);
+            currentTrigger.reset();
             return;
         }
     }
 }
 
-void WaitSet::detachAllConditions() noexcept
+void WaitSet::removeAllTrigger() noexcept
 {
-    for (auto& currentCondition : m_conditionVector)
+    for (auto& trigger : m_triggerVector)
     {
-        currentCondition->detachConditionVariable();
+        trigger.reset();
     }
-    m_conditionVector.clear();
+
+    m_triggerVector.clear();
 }
 
-WaitSet::ConditionVector WaitSet::timedWait(const units::Duration timeout) noexcept
+typename WaitSet::TriggerStateVector WaitSet::timedWait(const units::Duration timeout) noexcept
 {
     return waitAndReturnFulfilledConditions([this, timeout] { return !m_conditionVariableWaiter.timedWait(timeout); });
 }
 
-WaitSet::ConditionVector WaitSet::wait() noexcept
+typename WaitSet::TriggerStateVector WaitSet::wait() noexcept
 {
     return waitAndReturnFulfilledConditions([this] {
         m_conditionVariableWaiter.wait();
         return false;
     });
 }
 
-WaitSet::ConditionVector WaitSet::createVectorWithFullfilledConditions() noexcept
+typename WaitSet::TriggerStateVector WaitSet::createVectorWithFullfilledConditions() noexcept
 {
-    ConditionVector conditions;
-    for (auto& currentCondition : m_conditionVector)
+    TriggerStateVector conditions;
```

Comment:
don't call it Schnitzel. You removed everything that has Condition in its name and now the TriggerStates are conditions